src/components/HomeSection.test.jsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import HomeSection from './HomeSection';
+
+describe('HomeSection', () => {
+  test('checks if the title is in the document', () => {
+    render(
+      <HomeSection
+        title="Topology"
+        description="Math course"
+        buttons={[]}
+        pic="manifold"
+      />
+    );
+    screen.debug();
+    expect(screen.getByText('Topology')).toBeInTheDocument();
+  });
+
+  test('checks if the description is in the document', () => {
+    render(
+      <HomeSection
+        title="Topology"
+        description="Math course"
+        buttons={[]}
+        pic="manifold"
+      />
+    );
+    expect(screen.getByText('Math course')).toBeInTheDocument();
+  });
+
+  test('checks if it renders the given number of buttons', () => {
```

Comment:
```suggestion
  test('renders no buttons by default', () => {
```

Or better yet have a 'default render' test case that checks: no buttons are rendered and the title/description/pic are empty (you may need to set `defaultProps` for these in the component itself). Then, create another test case that checks the presence of one or more buttons, e.g. 'renders buttons given')